oneargopy/Argo.py

```diff
@@ -227,54 +345,432 @@ def __load_sprof_dataframe(self) -> pd:
+        # selected floats by pulling those floats from the 
+        # larger dataframes, only adding floats that match the 
+        # type to the frames. 
+        else:
+            if self.float_type != 'phys':
+                  # Make a list of bgc floats that the user wants 
+                  bgc_filter = (self.float_is_bgc_index['wmoid'].isin(self.floats)) & (self.float_is_bgc_index['is_bgc'] == True)
+                  selected_floats_bgc = self.float_is_bgc_index[bgc_filter]['wmoid'].tolist()
+                  # Gather bgc profiles for these floats from sprof index frame
+                  self.selected_from_sprof_index = self.sprof_index[self.sprof_index['wmoid'].isin(selected_floats_bgc)]
+            if self.float_type != 'bgc': 
+                  # Make a list of phys floats that the user wants 
+                  phys_filter = (self.float_is_bgc_index['wmoid'].isin(self.floats)) & (self.float_is_bgc_index['is_bgc'] == False)
+                  selected_floats_phys = self.float_is_bgc_index[phys_filter]['wmoid'].tolist()
+                  # Gather phys profiles for these floats from prof index frame
+                  self.selected_from_prof_index = self.prof_index[self.prof_index['wmoid'].isin(selected_floats_phys)]
+
+        if self.download_settings.verbose:
+            num_unique_floats = len(self.selected_from_sprof_index['wmoid'].unique()) + len(self.selected_from_prof_index['wmoid'].unique())
+            print(f"We will filter through {num_unique_floats} floats!") 
+            print(f'There are {len(self.selected_from_sprof_index) + len(self.selected_from_prof_index)} profiles are associated with these floats!')
+
+
+
+    def __narrow_profiles_by_criteria(self)-> dict:
+        """ A function to narrow down the available profiles to only those
+            that meet the criteria passed to select_profiles.
+
+            :return: narrowed_profiles : dict - A dictionary with float ID
+                keys corresponding to a list of profiles that match criteria.
+        """
+        # Filter by time, space, and type constraints first.
+        if self.float_type == 'bgc' : 
+            self.selection_frame_phys = pd.DataFrame()
+        else :
+            self.selection_frame_phys = self.__get_in_time_and_space_constraints(self.selected_from_prof_index)
+        if self.float_type == 'phys' : 
+            self.selection_frame_bgc = pd.DataFrame()
+        else :
+            self.selection_frame_bgc = self.__get_in_time_and_space_constraints(self.selected_from_sprof_index)
+        
+        # Set the selection frame
+        self.selection_frame = pd.concat([self.selection_frame_bgc, self.selection_frame_phys])
+
+        # Remove extraneous frames if keep index in memory is set to false
+        if not self.download_settings.keep_index_in_memory: 
+            del self.sprof_index
+            del self.prof_index
+            del self.selection_frame_bgc
```

Comment:
selection_frame_bgc and selection_frame_phys can always be deleted after concat, right?

Reply:
![image](https://github.com/lovelysandlonelys/OneArgo-Python-Dev/assets/91633941/348cbb83-8aa4-41ab-a382-99adee0ab9bf)